src/Backend/ToolchainImpl/DebianToolchain.ts

```diff
@@ -124,13 +106,32 @@ class DebianToolchain implements Toolchain {
     cmd.push("echo $?");
     return cmd;
   }
+}
+
+class DebianToolchain implements IToolchain<DebianTool> {
+  tool: DebianTool;
```

Comment:
Q. Why did you put install() or installed() commands inside `tool` not `toolchain`..? I wonder what is the intention of introducing `tool` because it looks good enough that `toolchain` has such functions.

Reply:
I mean it looks more natural that `toolchain` owns the installation-related functions, because it doens't own multiple tools that needs installation seperately.

Reply:
This design also has one tool. I just separate the install functions from other functions. I think `Toolchain` means `TRIV Toolchain`. And the way to install and uninstall the `TRIV Toolchain` varies from environment. So I don't think it is right that the two were tied together.

Reply:
Would it be better with install() get env as parameter or so? 
I just wondered you intention and I don't understand still... But it's a minor issue so I won't block this.